unittest/freedv_700d_comprx.c

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------*\
+
+  FILE........: freedv_700d_comprx.c
+  AUTHOR......: David Rowe
+  DATE CREATED: July 2022
+
+  Complex valued rx to support ctests.  Includes a few operations that will
+  only work if complex Tx and Rx signals are being handled correctly.
+ 
+\*---------------------------------------------------------------------------*/
+
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "freedv_api.h"
+#include "freedv_api_internal.h"
+#include "ofdm_internal.h"
+#include "codec2_cohpsk.h"
+#include "comp_prim.h"
+
+int main(int argc, char *argv[]) {
+
+    /* with no arguments then run with no test code */
+    int test_num = 0;
+    if (argc == 2) {
+        if (strcmp(argv[1],"tx") == 0) {
```

Comment:
Should this source file/executable be renamed to reflect the fact that it can do both TX and RX?

Reply:
This application only contains the FreeDV Rx `freedv_comprx()`, but includes test code to cover possible issues in the handing of the complex valued signals for both the FreeDv Tx and Rx .  If this was for general use - I'd split out the Tx/Rx test code into separate utilities, but for a specific test like this it's easier to do it all in one application.